CMSP-293-object-cache-pro-ea-docs (#8264)
pantheon-systems/documentation: source/content/guides/object-cache-pro/02-installing-configuring.md

### Requirements

Before you can install and activate Object Cache Pro, you must be working on a site with Redis enabled. You can activate this either from the Pantheon dashboard or with Terminus.

#### On the Dashboard

From your site's dashboard, click on the Settings button in the top right. In the modal window that comes up, you will see a box at the bottom that says "Redis". Click the "Enable" button.

![Pantheon Settings Add-Ons](../../../images/guides/object-cache-pro/pantheon-settings-addons.png)

#### With Terminus
To enable Redis via the command line with Terminus, use the following command:

```bash
terminus redis:enable <site>
```

### On regular WordPress Sites

#### 1. Install the Object Cache Pro plugin

Refer to the email you received and the link to the zip file for Object Cache Pro. Download the file, unzip, and add to your WordPress install as you normally would. 

*Note: This link will always point to the latest version of Object Cache Pro. Be sure to store it in a safe place so you never miss out on updates to Object Cache Pro.*

> Once you have Object Cache Pro in your codebase, there are some configuration options that you need to add to set it up. All of these options are stored in your `wp-config.php` in the root of your site repoistory.

Reviewer: ```suggestion
Once you have Object Cache Pro in your codebase, there are some configuration options that you need to add to set it up. All of these options are stored in your `wp-config.php` in the root of your site repository.
```

To add configuration values to Object Cache Pro for your site, you will need to add values to the `WP_REDIS_CONFIG` constant. On regular WordPress sites, this means saving these rules inside `wp-config.php`. If you are using a Bedrock-based upstream or WordPress (Composer Managed), you'll want to follow the instructions [below](#using-wordpress-composer-managed).

In `wp-config.php`, somewhere above the `/* That's all, stop editing! Happy Pressing. */` line at the bottom of the file, you'll want to add the following:

```php
/**
 * Object Cache Pro config
 */
define( 'WP_REDIS_CONFIG', [
    'token' => '<LICENSE-TOKEN>',
] );
```

Once this is done, there are additional options that need to be added. [Skip down to the configuration options](#configuring-object-cache-pro) section to finish the setup process.

### Using WordPress (Composer Managed)

*For full configuration instructions, refer to the [official Object Cache Pro documentation](https://objectcache.pro/docs/composer-installation).*

#### 1. Create the authentication token
First, you must add your license token to Composer. You can do this with the following command.

> `comoser config --auth http-basic.objectcache.pro token <LICENSE-TOKEN>`

Reviewer: ```suggestion
`composer config --auth http-basic.objectcache.pro token <LICENSE-TOKEN>`
```

> Alternately, you can create an `auth.json` file (created in the above command) manually:

Reviewer: ```suggestion
Alternately, you can create an `auth.json` file (created by the above command) manually:
```

The full, recommended contents of the `WP_REDIS_CONFIG` constant are as follows:

```php
	'token' => '<LICENSE-TOKEN>',
	'host' => $_SERVER['CACHE_HOST'] ?? '127.0.0.1',
	'port' => $_SERVER['CACHE_PORT'] ?? 6379,
	'database' => $_SERVER['CACHE_DB'] ?? 0,
	'password' => $_SERVER['CACHE_PASSWORD'] ?? null,
	'maxttl' => 86400 * 7,
	'timeout' => 1.0,
	'read_timeout' => 1.0,
	'debug' => false,
	'analytics' => [
		'enabled' => true,
		'persist' => true,
		'retention' => 3600, // 1 hour
		'footnote' => true,
	],
```

This should be added after `Config::define( 'WP_REDIS_CONFIG', [` in `config/application.php` for **WordPress (Composer Managed)**, or  `define( 'WP_REDIS_CONFIG', [` in `wp-config.php` for a **standard WordPress** site.

There are more [configuration options](https://objectcache.pro/docs/configuration-options) that can be added here as well. We recommend using the above settings which are optimized for Pantheon if you don't know specifically what each option does.

#### 2. Activate the plugin and enable Redis in the plugin

Before attempting to activate the plugin, make sure you `git push` your changes up to your repository.

> You can activate the Object Cache Pro plugin from the admin, locally with Lando or via Terminus. In the WordPress admin, navigate to the Plugins page and activate Object Cache Pro. Then go to the Object Cache Pro settings(?) page, ensure that you are in an environment with file write permissions (either SFTP mode, if activating on your Pantheon Dev environment or in a local development environment), and click (?) the Enable (?) button. This will create the `object-cache.php` drop-in file. If you did this in your Pantheon Dev environment, be sure to commit the change in SFTP mode. If you did this locally, you'll want to commit the file to your repository.

Reviewer: Do the question marks in this paragraph mean that we have to confirm the behavior? I know the path for OCP settings is `wp-admin/options-general.php?page=objectcache` and I think the others are correct based on [OCP docs](https://objectcache.pro/docs/installation#activation).

Author: Oh, good catch, yeah those were notes to myself for when I came back after testing! 🤣

		```php
		/**
		 * Object Cache Pro config
		 */
		define( 'WP_REDIS_CONFIG', [
			'token' => '<LICENSE-TOKEN>',
		] );
		```

1. Add Object Cache Pro configuration options after `define( 'WP_REDIS_CONFIG', [` in `wp-config.php` for a **standard WordPress** site. Note that there are more [configuration options](https://objectcache.pro/docs/configuration-options) that can be added. We recommend using the below settings which are optimized for Pantheon if you don't know specifically what each option does. The full, recommended contents of the `WP_REDIS_CONFIG` constant are:

		```php
		'token' => '<LICENSE-TOKEN>',
		'host' => $_SERVER['CACHE_HOST'] ?? '127.0.0.1',
		'port' => $_SERVER['CACHE_PORT'] ?? 6379,
		'database' => $_SERVER['CACHE_DB'] ?? 0,
		'password' => $_SERVER['CACHE_PASSWORD'] ?? null,
		'maxttl' => 86400 * 7,
		'timeout' => 1.0,
		'read_timeout' => 1.0,
		'debug' => false,
		'analytics' => [
			'enabled' => true,
			'persist' => true,
			'retention' => 3600, // 1 hour
			'footnote' => true,
			],
		```

1. Make sure you `git push` your changes up to your repository before you activate the plugin.

> 1. Activate the Object Cache Pro plugin and enable Redis in the plugin from the WordPress Admin, locally with Lando, or via Terminus.

Reviewer 2: @jazzsequence This step states that Lando can be used but no steps are provided. Should example steps for Lando be added or should the reference be removed?

Author: Can remove the reference to Lando.

Author: ```suggestion
1. Activate the Object Cache Pro plugin and enable Redis in the plugin from the WordPress Admin, locally with WP-CLI, or via Terminus.
```

Author: ```suggestion
1. Activate the Object Cache Pro plugin and enable Redis in the plugin from the WordPress Admin or via Terminus.
```

Reviewer 2: Thanks @jazzsequence . I added WP-CLI.

		```

		You can put this directly under the `WP_DEBUG` rules so it looks like this:

		```php
		/**
		 * Debugging Settings
		 */
		Config::define('WP_DEBUG_DISPLAY', false);
		Config::define('WP_DEBUG_LOG', false);
		Config::define('SCRIPT_DEBUG', false);
		ini_set('display_errors', '0');

		/**
		 * Object Cache Pro config
		 */
		Config::define( 'WP_REDIS_CONFIG', [
			'token' => '<LICENSE-TOKEN>',
		] );
		```

1. Add Object Cache Pro configuration options after `Config::define( 'WP_REDIS_CONFIG', [` in `config/application.php` for **WordPress (Composer Managed)** sites. The full, recommended contents of the WP_REDIS_CONFIG constant are:

		```php
		'token' => '<LICENSE-TOKEN>',
		'host' => $_SERVER['CACHE_HOST'] ?? '127.0.0.1',
		'port' => $_SERVER['CACHE_PORT'] ?? 6379,
		'database' => $_SERVER['CACHE_DB'] ?? 0,
		'password' => $_SERVER['CACHE_PASSWORD'] ?? null,
		'maxttl' => 86400 * 7,
		'timeout' => 1.0,
		'read_timeout' => 1.0,
		'debug' => false,
		'analytics' => [
			'enabled' => true,
			'persist' => true,
			'retention' => 3600, // 1 hour
			'footnote' => true,
		],
		```
1. Make sure you `git push` your changes up to your repository before you activate the plugin.

> 1. Activate the plugin and enable Redis in the plugin. You can activate the Object Cache Pro plugin from the WordPress Admin, locally with Lando, or via Terminus.

Author: Samesies.

> 1. Activate the plugin and enable Redis in the plugin. You can activate the Object Cache Pro plugin from the WordPress Admin, locally with Lando, or via Terminus.

Author: ```suggestion
1. Activate the plugin and enable Redis in the plugin. You can activate the Object Cache Pro plugin from the WordPress Admin, locally with WP-CLI, or via Terminus.
```

> 1. Activate the plugin and enable Redis in the plugin. You can activate the Object Cache Pro plugin from the WordPress Admin, locally with Lando, or via Terminus.

Author: ```suggestion
1. Activate the plugin and enable Redis in the plugin. You can activate the Object Cache Pro plugin from the WordPress Admin or via Terminus.
```
